### ISSUE TITLE
Failure to build examples on Arch linux

### ISSUE THREAD

**Issue by justinmoon** [revery-ui/revery#768]
I installed everything mentioned on the [build instructions](https://github.com/revery-ui/revery/wiki/Building-&-Installing) and got this error:

```
$ esy x Examples
...
error: build failed with exit code: 1
  build log:
    # esy-build-package: building: esy-nasm@github:revery-ui/esy-nasm#64a802b
    # esy-build-package: pwd: /home/justin/.esy/3/b/esy_nasm-f2002482
    # esy-build-package: running: 'bash' 'esy-build.sh' 'linux'
    Error: nasm is not installed
    error: command failed: 'bash' 'esy-build.sh' 'linux' (exited with 1)
    esy-build-package: exiting with errors above...
```

Installing [`nasm`](https://www.archlinux.org/packages/extra/x86_64/nasm/) fixed the problem.

**Comment by Et7f3** [comment]
Is it weird I have managed to build revery inside a arch docker with the instruction I wrote.

**Comment by justinmoon** [comment]
Perhaps you already had nasm / clang installed? The instructions don't mention them anywhere, and both seem to be required to build the examples.

**Comment by Et7f3** [comment]
Do you see `esy-nasm` in the esy.lock folder ?

**Comment by justinmoon** [comment]
No. This is what my `esy.lock` folder looks like: https://gist.github.com/justinmoon/ce76e4635529b837524a19cc609c0569

**Comment by Et7f3** [comment]
Can you send also esy.lock/index.json

**Comment by justinmoon** [comment]
Here's the [index.json](https://github.com/revery-ui/revery/files/4202613/index.txt)

**Comment by Et7f3** [comment]
Because we make sure it is available for you: https://github.com/revery-ui/revery/blob/master/esy.lock/index.json#L315-L326.
esy-nasm is required by [`esy-libjpeg-turbo`](https://github.com/revery-ui/revery/blob/master/esy.lock/index.json#L338)
~~can you clone it: https://github.com/revery-ui/libjpeg-turbo, remove nasm, run `esy` inside. Does it build ? If no can you send the error.~~

~~It is very strange because: I have a arch without nasm and it build.~~

Ok it is because we have added support for CentOS so we don't download nasm.

Thanks it is modified.

**Comment by Et7f3** [comment]
@bryphe What should we do ? Should we mix the old with the new bahaviour ? https://github.com/revery-ui/esy-nasm/commit/64a802b70c403634d8f9698bb85563932f7c0cfa

**Comment by bryphe** [comment]
@Et7f3 - the best fix would be to get `esy-nasm` building across CentOS and all Linux - fix whatever issues are blocking it - then we have it in our `esy` sandbox and don't have to think about it.

Otherwise, we should add it to the docs as a dependency.

**Comment by zicklag** [comment]
I'm getting the same issue on Ubuntu 20.04, but I have clang and nasm in my PATH on my host and it's still saying that `nasm` isn't intalled.  How did you get it working @justinmoon ?

**Comment by kyldvs** [comment]
@zicklag I just went through the process of figuring out how to get Revery building on Ubuntu 20.04 last night, this is what I had to do:

1. Install all the packages listed here: https://github.com/revery-ui/revery/wiki/Building-&-Installing#for-linux-users
2. Make sure you're on a recent version of esy
3. Then I used this configuration for package.json: https://github.com/kyldvs/template-revery-app

(step 1 installing `nasm` fixed the missing nasm issue for me)
